twine/__main__.py

```diff
@@ -13,40 +13,57 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 import http
+import logging
 import sys
 from typing import Any
 
-import colorama
 import requests
+import rich.console
+import rich.highlighter
+import rich.logging
+import rich.theme
 
 from twine import cli
 from twine import exceptions
 
 
 def main() -> Any:
+    root_logger = logging.getLogger("twine")
+    root_logger.addHandler(
+        rich.logging.RichHandler(
+            console=rich.console.Console(
+                theme=rich.theme.Theme(
+                    {
+                        "logging.level.debug": "green",
+                        "logging.level.info": "blue",
+                        "logging.level.warning": "yellow",
+                        "logging.level.error": "red",
+                        "logging.level.critical": "reverse red",
+                    }
+                )
+            ),
+            show_time=False,
+            show_path=False,
+            highlighter=rich.highlighter.NullHighlighter(),
+        )
+    )
```

Comment:
@sigmavirus24 This replaces colorama for top-level error reporting. What do you think?

<img width="675" alt="Screen Shot 2022-01-02 at 6 32 30 AM" src="https://user-images.githubusercontent.com/1326704/147874477-03908860-8942-4574-b22a-b77ad07f9af2.png">